### DISCUSSION TITLE
better docs for all types of operations

### DISCUSSION THREAD

**Discussion by Kubas556** [dokan-dev/dokan-dotnet#361] [Q&A] [answered]
Hi, is somewhere better explained lifecycle of each operation (reading file, writing file, etc.)? I know nothing about filesystem and is almost impossible to write something functional with this library and debugging it is very painful. I can only copy paste from example code.

**Comment by Liryna** [comment]
Hi @Kubas556 ,

Dokan-dotnet is just a wrapper of the C library. You can look at https://dokan-dev.github.io/dokany-doc/html/ that explains the workflow of the calls. Otherwise, Dokan just transfers what the system is requesting so Microsoft documentation is really the core documentation to look at and understand how it works.

**Reply by Kubas556** [reply]
do you have link to microsoft docs?

**Reply by Liryna** [reply, marked as answer]
If you are interested in CreateFile you can look [here](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilea), or [WriteFile](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-writefile) etc

**Reply by Kubas556** [reply]
Thanks!
